okapi-core/src/test/java/org/folio/okapi/MultiTenantTest.java

```diff
@@ -142,15 +143,19 @@ private void td(TestContext context, Async async) {
   public void tearDown(TestContext context) {
     Async async = context.async();
     HttpClient httpClient = vertx.createHttpClient();
-
-    httpClient.delete(port, "localhost",
-      "/_/discovery/modules", context.asyncAssertSuccess(response -> {
-      context.assertEquals(204, response.statusCode());
-      response.endHandler(x -> {
-        httpClient.close();
-        td(context, async);
-      });
-    }));
+    httpClient.request(HttpMethod.DELETE, port,
+        "localhost", "/_/discovery/modules", context.asyncAssertSuccess(request -> {
+          request.end();
+          request.onComplete(context.asyncAssertSuccess(response -> {
+            context.assertEquals(204, response.statusCode());
+            response.endHandler(x -> {
+              httpClient.close();
+              vertx.close(y -> {
+                async.complete();
+              });
```

Comment:
This is a more natural way because it also checks that closing vertx succeeds:

```
httpClient.close();
vertx.close(context.asyncAssertSuccess());
async.complete();
```